docs/js-guide.md

```diff
@@ -33,17 +36,58 @@ Projects may use any ES language features which meet all of the following criter
 As an exception, [JSX](https://reactjs.org/docs/introducing-jsx.html) is used for
 creating UI components.
 
+### Module naming
+
+Modules should use lowercase, hyphen-separated names (`helpful-module.js`).
+
+**Exception:** Modules whose primary export is a Preact component should use a
+CamelCase name that matches the component name (eg. `ShinyButton.js`). _This is a
+React/Preact ecosystem convention._
+
+### Import structure
+
+Imports should be placed at the top of a module, grouped into sections by their
+relation to the current project:
+
+- Platform (node, browser) imports
+- Third-party imports
+- Imports from other directories in the current package
+- Imports from the same directory in the current package
+
+Within each group import lines should be sorted alphabetically by module path.
+
+```js
+import { readFile } from 'fs';
+
+import { render } from 'markdown';
+
+import { fetchData } from './data-source';
+import { formatDate } from './utils/date';
+```
+
+### Default exports
+
+Modules should always use named exports rather than default exports.
```

Comment:
I wonder if we might avoid using the word _always_ in a convention/rule?

Reply:
I've reworded this sentence in a way that avoids "always" and is hopefully clearer.